fog/view/enclave/impl/src/lib.rs

```diff
@@ -182,6 +193,29 @@ where
         for rec in records {
             store.add_record(&rec.search_key, &rec.payload)?;
         }
+
         Ok(())
     }
+
+    /// Takes in a client's query request and returns a list of query requests
+    /// to be sent off to each Fog View Store shard.
+    fn create_multi_view_store_query_data(
+        &self,
+        client_query: EnclaveMessage<ClientSession>,
+    ) -> Result<Vec<EnclaveMessage<ClientSession>>> {
+        let client_query_bytes = self.ake.client_decrypt(client_query.clone())?;
+
+        let mut encryptors = self.store_encryptors.lock()?;
+        let mut results = Vec::new();
+        for store_encryptor in encryptors.deref_mut() {
+            let data = store_encryptor.encrypt(&client_query.aad, &client_query_bytes)?;
+            results.push(EnclaveMessage {
+                aad: client_query.clone().aad,
+                channel_id: client_query.clone().channel_id,
+                data,
+            });
```

Comment:
Let's clone the specific fields we want, rather than cloning the query twice.
```suggestion
            let aad = client_query.aad.clone();
            let data = store_encryptor.encrypt(&aad, &client_query_bytes)?;
            let channel_id = client_query.channel_id.clone();
            results.push(EnclaveMessage { aad, channel_id, data });
```

Reply:
Done.